### ISSUE TITLE
lotw issue with DXCC_QSLs 

### ISSUE THREAD

**Issue by otalado** [n1kdo/lotw-qso-analyzer#8]
I apologize for bothering you since I am almost sure there is no issue with your code. As I am a total newbie to LotW, I hope I don't ask too much since you have a much better understanding of all things LotW.

When I issued the  QSL card requests manually, I got the following response:
```
ARRL Logbook of the World DXCC QSL Card Report
Generated at 2024-11-27 05:31:38Z for S53C

<ADIFVER:5>3.0.5
<PROGRAMID:17>LoTW qslcards.php
<PROGRAMVERSION:6>v1.5.3

<APP_LoTW_DXCCRecord_Call:S>
<APP_LoTW_DXCCRecord_Entity:3:N>499
<APP_LoTW_DXCCRecord_Country:8:S>SLOVENIA
<APP_LoTW_DXCCRecord_Name:S>
<APP_LoTW_DXCCRecord_ARRLId:S>
<APP_LoTW_DXCCRecord_SysId:S>
<APP_LoTW_DXCCRecord_LoTWId:6:N>443183
<APP_LoTW_DXCCRecord_Updated:S>
<APP_LoTW_NUMREC:1:N>0

<eoh>
```
Some items returned are plain wrong, i.e. <APP_LoTW_DXCCRecord_Call:S> should be **<APP_LoTW_DXCCRecord_Call:4:S>S53C** and so for.  When I try to get the DXCC QSL cards ADIF in your program the same response is received and naturally the program crashes in adif.py at the adif_field(s) function.  I tried using an uppercase and lowercase callsign, just to be sure, but there was no difference.
I don't have any QSLs as I have uploaded just a few 'fake' QSOs for testing. Is it possible that LotW does not handle that situation correctly??

**Comment by n1kdo** [comment]
I was able to pull both LoTW QSO/QSL records and DXCC qslcards records today, no problems.

When you mention calling the cards API manually, how did you do that?  

Do you have a DXCC record at ARRL?  Otherwise, there are no cards to be had.

I  did a query like this:

GET https://lotw.arrl.org/lotwuser/logbook/qslcards.php?login=n1kdo&password=(REDACTED)&ac_acct=1

note the ac_acct=1 parameter -- which is undocumented!  
(ref https://lotw.arrl.org/lotw-help/developer-query-dxcc-credits/?lang=en)

the document returned starts with this header:

```
ARRL Logbook of the World DXCC QSL Card Report
Generated at 2024-11-28 15:46:25Z for N1KDO

<ADIFVER:5>3.0.5
<PROGRAMID:17>LoTW qslcards.php
<PROGRAMVERSION:6>v1.5.3

<APP_LoTW_DXCCRecord_Call:5:S>N1KDO
<APP_LoTW_DXCCRecord_Entity:3:N>291
<APP_LoTW_DXCCRecord_Country:24:S>UNITED STATES OF AMERICA
<APP_LoTW_DXCCRecord_Name:19:S>Jeffrey B. Otterson
<APP_LoTW_DXCCRecord_ARRLId:10:S>0001609529
<APP_LoTW_DXCCRecord_SysId:7:S>00013HN
<APP_LoTW_DXCCRecord_LoTWId:5:N>25861
<APP_LoTW_DXCCRecord_Updated:19:S>2023-05-25 05:32:23
<APP_LoTW_NUMREC:4:N>1845

<eoh>

```

In short, I think that ARRL is sending bad data, and there's not a lot I can do about that.

I think you need to submit a bug fix request to the LoTW "development team" (Which I believe is Jon Bloom...)

If you would send me the error that the adif parser emits, I can fix that so it will not crash, but until the source data works right, there is little else I can do.

73

Jeff

**Comment by n1kdo** [comment]
If I answered this to your satisfaction I would appreciate it if you would close the issue.  Thanks.

**Comment by otalado** [comment]
Thanks for your reply. I suspected something was wrong with a LoTW, but since I have no experience with it, I was not sure.
About issuing the API request 'manually', I expressed myself the wrong way. What I meant is that I issued a request in the browser like this: https://lotw.arrl.org/lotwuser/logbook/qslcards.php?login=s53c&password=****&ac_acct=1. I did not know that ac_acct=1 is undocumented (I just picked it from your code). No need to fix your code, I can do it myself.

73 and thanks again!

**Comment by otalado** [comment]
No need for action
